lib/features/email/presentation/widgets/calendar_event/calendar_event_action_button_widget.dart

```diff
@@ -3,54 +3,67 @@ import 'package:core/core.dart';
 import 'package:flutter/material.dart';
 import 'package:get/get.dart';
 import 'package:tmail_ui_user/features/email/domain/model/event_action.dart';
+import 'package:tmail_ui_user/features/email/presentation/controller/single_email_controller.dart';
 import 'package:tmail_ui_user/features/email/presentation/styles/calendar_event_action_button_widget_styles.dart';
-import 'package:tmail_ui_user/main/utils/app_utils.dart';
+
+typedef OnCalendarEventReplyActionClick = void Function(EventAction eventAction);
 
 class CalendarEventActionButtonWidget extends StatelessWidget {
 
   final List<EventAction> eventActions;
   final EdgeInsetsGeometry? margin;
+  final OnCalendarEventReplyActionClick onCalendarEventReplyActionClick;
 
   const CalendarEventActionButtonWidget({
     super.key,
     required this.eventActions,
+    required this.onCalendarEventReplyActionClick,
     this.margin,
   });
 
   @override
   Widget build(BuildContext context) {
     final responsiveUtils = Get.find<ResponsiveUtils>();
+
     return Container(
       width: double.infinity,
       margin: margin ?? CalendarEventActionButtonWidgetStyles.margin,
       padding: responsiveUtils.isPortraitMobile(context)
         ? CalendarEventActionButtonWidgetStyles.paddingMobile
         : CalendarEventActionButtonWidgetStyles.paddingWeb,
-      child: Wrap(
-        spacing: CalendarEventActionButtonWidgetStyles.space,
-        runSpacing: CalendarEventActionButtonWidgetStyles.space,
-        children: eventActions
-          .map((action) => TMailButtonWidget(
-            text: action.actionType.getLabelButton(context),
-            backgroundColor: CalendarEventActionButtonWidgetStyles.backgroundColor,
-            borderRadius: CalendarEventActionButtonWidgetStyles.borderRadius,
-            padding: CalendarEventActionButtonWidgetStyles.buttonPadding,
-            textStyle: const TextStyle(
-              fontWeight: CalendarEventActionButtonWidgetStyles.fontWeight,
-              fontSize: CalendarEventActionButtonWidgetStyles.textSize,
-              color: CalendarEventActionButtonWidgetStyles.textColor,
-            ),
-            textAlign: TextAlign.center,
-            minWidth: CalendarEventActionButtonWidgetStyles.minWidth,
-            width: responsiveUtils.isPortraitMobile(context) ? double.infinity : null,
-            border: Border.all(
-              width: CalendarEventActionButtonWidgetStyles.borderWidth,
-              color: CalendarEventActionButtonWidgetStyles.textColor
-            ),
-            onTapActionCallback: () => AppUtils.launchLink(action.link),
-          ))
-          .toList(),
-      ),
+      child: Obx(() {
+        final controller = Get.find<SingleEmailController>();
+        
+        return Wrap(
```

Comment:
I don't think this is a good solution. It may crash when `SingleEmailController` is deleted for some reason.

Reply:
What do you suggest?

Reply:
Why not pass it as a property?

Reply:
> Propose: Pass the object to listen as a property of the class

The calendar buttons live inside `EmailView`. If I do this, whenever those buttons' UI change, the whole `EmailView` will be rebuilt. I'm simply isolating the rebuild scope of them.

Reply:
> > Propose: Pass the object to listen as a property of the class
> 
> The calendar buttons live inside `EmailView`. If I do this, whenever those buttons' UI change, the whole `EmailView` will be rebuilt. I'm simply isolating the rebuild scope of them.

Are you sure? Depending on what variable you listen for, it will rebuild like that. We have many ways to just rebuild that widget. Please do more research.

Reply:
Done